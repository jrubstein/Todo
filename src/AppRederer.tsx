import * as React from 'react'
import {renderToString} from 'react-dom/server'
import { JssProvider, SheetsRegistry } from 'react-jss'
import TodoContainer from './views/components/TodoContainer';
import { Provider } from 'react-redux';
import { createNewStore, initialState } from './views/reducers/Store'
import { sheet } from './views/styles/base';

const template = (app) => `
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <title>Todos</title>
    <style type="text/css" id="styles">
      ${sheet.toString()}
    </style>
  </head>
  <body>
    <div id="main-container">${app}</div>
    <script src="/js/app.js" type="text/javascript"></script>
  </body>
</html>
`

// Render the page template.
export const renderIndex = () => {
    const registry = new SheetsRegistry()
    const app = renderToString(
        <JssProvider registry={registry}>
            <Provider store={createNewStore(initialState)}>
              <TodoContainer />
            </Provider>
        </JssProvider>
      )
    return template(app)
}