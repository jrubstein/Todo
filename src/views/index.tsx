import * as React from 'react'
import * as ReactDOM from 'react-dom'
import { Provider } from 'react-redux'

import { createNewStore } from './reducers/Store'
import TodoContainer from './components/TodoContainer'
import { initialState, State } from './reducers/Store';

// Hydrate the markup from the server.
ReactDOM.hydrate(
    <Provider store={createNewStore(initialState)}>
        <TodoContainer />
    </Provider>,
    document.getElementById('main-container')
)