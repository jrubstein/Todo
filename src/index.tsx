import * as path from 'path'
import * as Koa from 'koa'
import * as serve from 'koa-static'
import jss from 'jss'
import preset from 'jss-preset-default'
import { renderIndex } from './AppRederer'

jss.setup(preset())

const app = new Koa()
app.use(serve(path.join(__dirname, '..', '..', 'public')))

// Returns only the Todo list
app.use(ctx => {
    ctx.body = renderIndex()
})

app.listen(process.env.PORT || 3000, () => {
    console.log('started')
})
