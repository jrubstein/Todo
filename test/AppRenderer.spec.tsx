import { renderIndex } from "../src/AppRederer";

describe('App renderer test', () => {
    it('Should render the template', () => {
        const page = renderIndex()
        expect(page).toMatchSnapshot()
    })
    it('Should have the app.js file', () => {
        const page = renderIndex()
        expect(page.indexOf('/js/app.js') > 0).toBeTruthy()
    })
})