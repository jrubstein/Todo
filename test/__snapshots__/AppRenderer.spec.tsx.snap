// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renderer test Should render the template 1`] = `
"
<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\">
    <title>Todos</title>
    <style type=\\"text/css\\" id=\\"styles\\">
      .container-0-0-1 {
  max-width: 800px;
  margin: auto;
  text-align: center;
}
.numberOfItems-0-0-2 {
  padding: 16px;
}
.item-0-0-3 {
  padding: 8px;
}
.date-0-0-5 {
  font-size: 10px;
  padding: 0 8px;
  color: gray;
}
.icon-0-0-6 {
  background-repeat: no-repeat;
  background-size: 18px 18px;
  width: 18px;
  height: 18px;
  border: none;
  padding: 0 12px;
}
.deleteButton-0-0-7 {
  background-image: url('icons/delete.svg');
}
.editButton-0-0-8 {
  background-image: url('icons/edit.svg');
}
.disabledEditButton-0-0-9 {
  opacity: 0.3;
}
.saveButton-0-0-10 {
  background-image: url('icons/save.svg');
}
.todoInput-0-0-11 {
  display: flex;
  justify-content: space-evenly;
  width: 300px;
  margin: auto;
  padding: 16px;
}
.todoTextInput-0-0-12 {
  height: 14px;
  border-bottom: 1px solid blue;
  border-top: 0;
  border-left: 0;
  border-right: 0;
}
    </style>
  </head>
  <body>
    <div id=\\"main-container\\"><div class=\\"container-0-0-1\\" data-reactroot=\\"\\"><h1>Todos</h1><h2 class=\\"numberOfItems-0-0-2\\">0<!-- --> Item(s)</h2><form class=\\"todoInput-0-0-11\\"><label for=\\"todoInput\\">Add todo here</label><input type=\\"text\\" class=\\"todoTextInput-0-0-12\\" id=\\"todoInput\\" required=\\"\\"/><button class=\\"icon-0-0-6 saveButton-0-0-10\\" type=\\"submit\\" aria-label=\\"save\\"></button></form></div></div>
    <script src=\\"/js/app.js\\" type=\\"text/javascript\\"></script>
  </body>
</html>
"
`;
